portal/server/routes/admin/users.js

```diff
@@ -61,22 +61,38 @@ const handleRoles = (roles) => {
 
 // Admin - user create
 router.post('/users/create', async (req, res) => {
-  const { firstname, surname, roles } = req.body;
-
-  if (firstname && surname && roles) {
+  const {
+    firstname,
+    surname,
+    roles,
+    email,
+    bank,
+  } = req.body;
+
+  if (firstname && surname && roles && bank) {
     const { userToken } = requestParams(req);
-    const userToCreate = {
+    const user = {
       ...req.body,
+      username: email,
       roles: handleRoles(req.body.roles),
     };
 
     // inflate the bank object
     const banks = await getApiData(api.banks(userToken), res);
 
-    const selectedBank = banks.find((bank) => bank.id === userToCreate.bank);
-    userToCreate.bank = selectedBank;
-    userToCreate.username = userToCreate.email;
-    const { status, data } = await api.createUser(userToCreate, userToken);
+    // `fi` stands for `financial institution`
+    if (bank === 'all') {
+      user.bank = {
+        id: '*',
+        name: 'All',
+        hasGefAccessOnly: false,
+      };
```

Comment:
Is this new functionality we're adding?

Reply:
This was missing, since there was schema validation before a non-existent `bank` object was not being checked.

Reply:
Is it a problem that we've saving this new bank data to the database now though?